plugins/destination/mongodb/main.go

```diff
@@ -1,17 +1,24 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
+
 	"github.com/cloudquery/cloudquery/plugins/destination/mongodb/client"
-	"github.com/cloudquery/cloudquery/plugins/destination/mongodb/resources/plugin"
-	"github.com/cloudquery/plugin-sdk/v3/plugins/destination"
-	"github.com/cloudquery/plugin-sdk/v3/serve"
+	internalPlugin "github.com/cloudquery/cloudquery/plugins/destination/mongodb/resources/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/serve"
 )
 
 const (
 	sentryDSN = "https://f4351c889aae4dde9b87a916d44ce836@o1396617.ingest.sentry.io/4504374724657152"
 )
 
 func main() {
-	p := destination.NewPlugin("mongodb", plugin.Version, client.New, destination.WithManagedWriter())
-	serve.Destination(p, serve.WithDestinationSentryDSN(sentryDSN))
+	p := plugin.NewPlugin("mongodb", internalPlugin.Version, client.New)
+	if err := serve.Plugin(p, serve.WithPluginSentryDSN(sentryDSN), serve.WithDestinationV0V1Server()).Serve(context.Background()); err != nil {
+		fmt.Println(err)
```

Comment:
nit:
```suggestion
				log.Fatalf("failed to serve plugin: %v", err)
```
then no need for os.Exit(1) below.